### DISCUSSION TITLE
Add automatic updates to RustDesk and update in app

### DISCUSSION THREAD

**Discussion by AdanC1b3r** [rustdesk/rustdesk#5146] [Ideas]
In the recent update I noticed that they have not yet added automatic software updates, for some users it is tedious to download the installer from the website or the project on GitHub every time they update, if they find the option to update in the software it would be a success.

**Comment by dinger1986** [comment]
This would be good and very helpful

**Comment by bovirus** [comment]
@rustdesk 

I believe that can be an useful function.

**Reply by rustdesk** [reply]
I guess nobody do not know it is useful in this world. :)

**Comment by bovirus** [comment]
@rustdesk

A good advertising say that the tecnology should make simple what is quite complicated.

Then if you can enable the automatic update could be a good idea.

Because I found probably a bug in Windows if you try to update a program remotely teh application inr emote desktop.

If you have example 1.2.1 installed in remote client and tray icon icon active when I download and run the night build 1.2.2 in the remote client when I go to the settings about program shorcut craetion /install certificate/icon on desktop the are become gary and I cannot rpoecd.
I can just close the instalaltion without update the program.

**Reply by rustdesk** [reply]
Please do not always @ me, I do have very less time to write code because so many @ me everyday. :)

**Comment by bovirus** [comment]
Sorry. ;-)

**Comment by claell** [comment]
@rustdesk Sorry for the add, but hopefully a quick solution until auto update is available: Some hint that a new version is available (maybe with a link where to download it) will already be rather helpful, while not that hard to implement, I guess.

**Reply by claell** [reply]
Ah, seems to be discussed in https://github.com/rustdesk/rustdesk/discussions/5475.

**Reply by xlionjuan** [reply]
You could use some trick to upgrade RustDesk more easily, though it isn't automatic.

## On macOS you can use [Homebrew](https://brew.sh):
```bash
brew upgrade rustdesk
```

>Note: You need to use homebrew to install an application once before in order to upgrade it, or it will return error said ```RustDesk.app``` is already exists, the solution is manually delete ``RustDesk.app`` from your system's ``Applications`` folder and run this command again.

## On Windows you could use winget:
```bash
winget install rustdesk.rustdesk
```
> Just run this on Terminal/PowerShell with administrator
<hr>

No convenience method for Linux now, they should host their repository server for RPMs/APT.

**Reply by xlionjuan** [reply]
**Unofficial** APT/RPM repo: Check my profile 

Android: F-Droid Version, but the code signing is different from GitHub version, it means you need to reinstall

**Reply by Abejo** [reply]
Hi, thanks for the reply, the question was mostly for the Windows client, since most people don't update unless it's automatic.

**Comment by Abejo** [comment]
Has there been any progress with this? Is there a possibility that automatic updates will be available in the future?

**Comment by sirocyl** [comment]
On Windows, I usually use WinGet for application updates - it would be neat if the application would see whether it had been installed using WinGet, and if so, check for and install the update through that.

That would require that the WinGet manifest is up to date, though, I don't think the automation has been run for the latest version yet (see https://github.com/microsoft/winget-pkgs/issues/194971 and previous RustDesk updates: https://github.com/microsoft/winget-pkgs/commits?author=rustdesk)

**Comment by ProvaTeams** [comment]
On Windows it would be possible to download the MSI in %temp%, run it with the /quiet parameter, and delete the download from temp

**Reply by xlionjuan** [reply]
On Windows just use `winget`

**Reply by Abejo** [reply]
On Windows LTSC winget is not installed, it can be installed, but the idea is to simplify things.

**Reply by sirocyl** [reply]
winget would be fine imo if it were itself installed using winget - but if the installer came from another source, or if I didn't want to use winget, I don't think programs should take it over - especially since the first invocation and use of winget has some agreements and terms, and makes some changes to the system itself.

**Reply by xlionjuan** [reply]
**It depends on the software and the manifest**, on RustDesk, unless you're using Custom client or msi methods, `winget` will be fine, because the `winget` version of RustDesk is just GitHub's `exe` version and add *silent install parameter*, it is same as you downloaded `exe` and manually installed it.
